test/manual/events/src/providers/video-element-provider.ts

```diff
@@ -0,0 +1,673 @@
+                this.videoElement.style.objectFit = 'fill';
+            }
+        }
+    }
+
+    getContainer(): HTMLDivElement | null {
+        return this.container;
+    }
+
+    setContainer(element: HTMLDivElement): void {
+        this.container = element;
+        if (element && this.videoElement.parentNode !== element) {
+            element.appendChild(this.videoElement);
+        }
+    }
+
+    removeFromContainer(): void {
+        const { container, videoElement } = this;
+        this.container = null;
+        if (container && container === videoElement.parentNode) {
+            container.removeChild(videoElement);
+        }
+    }
+
+    public remove(): void {
+        const container = this.container;
+        const video = this.videoElement;
+        this.stop();
+        this.destroy();
+        if (container) {
+            container.removeChild(video);
+        }
+    }
+
+    public stop(): void {
+        this.seeking = false;
+        if (this.videoElement) {
+            this.videoElement.removeAttribute('preload');
+            this.videoElement.removeAttribute('src');
+            this.videoElement.load();
+        }
+        this.setState('idle');
+    }
+
+    public destroy(): void {
+        this.off();
+        this.detachMedia();
+        this.item = null;
+        this.seeking = false;
+        this.container = null;
+        // @ts-ignore
+        this.config = null;
+        // @ts-ignore
+        this.videoElement = null;
+    }
+
+    public supportsFullscreen(): boolean {
+        return true;
+    }
+
+    public setVisibility(state): void {
+        state = !!state;
+        if (this.container) {
+            this.container.style.opacity = state ? '1' : '0';
+        }
+    }
+
+    public play(): Promise<void> {
+        return this.videoElement.play();
+    }
+
+    public pause(): void {
+        this.videoElement.pause();
+    }
+
+    public seek(toPosition: number) {
+        const seekRange = this.getSeekRange();
+        if (toPosition < 0) {
+            this.seekToTime = toPosition + seekRange.end;
+        } else {
+            this.seekToTime = toPosition;
+        }
+        if (!this.getSeekableEnd()) {
+            // Can't seek without seekable range, trigger playback
+            // TODO: Defer seeking until seekable range updates
+            this.item.starttime = this.seekToTime;
+            this.play();
+            return;
+        }
+
+        this.seeking = true;
+        this.seekFromTime = this.videoElement.currentTime;
+        this.videoElement.currentTime = this.seekToTime;
+    }
+
+    public getPlaybackRate(): number {
+        return this.videoElement.playbackRate;
+    }
+
+    public setPlaybackRate(playbackRate: number) {
+        this.videoElement.playbackRate = this.videoElement.defaultPlaybackRate = playbackRate;
+    }
+
+    public getCurrentQuality(): number {
+        return 0;
+    }
+
+    public setCurrentQuality(qualityLevel: number): void {
```

Comment:
expected implementation?

Reply:
Added a comment. Not implementing since there is no way to select quality on a video element for `video.src =` playback in any browser.